File: Udayland.cpp
#include <bits/stdc++.h>
using namespace std;

int main(void) {
    int n;
    cin >> n;
    vector<string> vs(n);
    for (auto &x:vs)
        cin >> x;
    bool state = false;
    bool in = true;
    for (int i=0; i<n; i++) {
        if (in) {
            for (int j=0; j<4; j++) {
                if (vs[i][j]=='O' && vs[i][j+1]=='O') {
                    state = true;
                    vs[i][j] = '+';
                    vs[i][j+1] = '+';
                    in = false;
                    break;
                }
            }
        }
        else 
            break;  
    }
    if (state) {
        cout << "YES\n";
        for (auto &x:vs)
            cout << x << endl;
    }
    else 
        cout << "NO\n";

    return 0;
}

File: Python/Udayland.py
def main():
    n = int(input())
    vs = [input() for i in range(5)]
    state = False
    inn = True
    for i in range(n):
        if inn:
            for j in range(4):
                if vs[i][j] == "O" and vs[i][j+1] == "O":
                    state = True
                    vs[i][j] = "+"
                    vs[i][j+1] = "+"
                    inn = False
                    break
        else:
            break
    if state:
        print("YES")
        for i in vs:
            print(i)
    else:
        print("NO")


if __name__ == '__main__':
    main()
